eclair-core/src/test/scala/fr/acinq/eclair/integration/basic/fixtures/MinimalNodeFixture.scala

```diff
@@ -0,0 +1,272 @@
+import fr.acinq.eclair.blockchain.DummyOnChainWallet
+import fr.acinq.eclair.blockchain.bitcoind.ZmqWatcher
+import fr.acinq.eclair.blockchain.bitcoind.ZmqWatcher.{WatchFundingConfirmed, WatchFundingConfirmedTriggered, WatchFundingDeeplyBuried, WatchFundingDeeplyBuriedTriggered}
+import fr.acinq.eclair.channel.ChannelOpenResponse.ChannelOpened
+import fr.acinq.eclair.channel._
+import fr.acinq.eclair.channel.fsm.Channel
+import fr.acinq.eclair.crypto.TransportHandler
+import fr.acinq.eclair.crypto.keymanager.{LocalChannelKeyManager, LocalNodeKeyManager}
+import fr.acinq.eclair.io.PeerConnection.ConnectionResult
+import fr.acinq.eclair.io.{Peer, PeerConnection, Switchboard}
+import fr.acinq.eclair.payment.receive.{MultiPartHandler, PaymentHandler}
+import fr.acinq.eclair.payment.relay.Relayer
+import fr.acinq.eclair.payment.send.PaymentInitiator
+import fr.acinq.eclair.payment.{Bolt11Invoice, PaymentSent}
+import fr.acinq.eclair.router.Router
+import fr.acinq.eclair.wire.protocol.IPAddress
+import fr.acinq.eclair.{BlockHeight, MilliSatoshi, MilliSatoshiLong, NodeParams, ShortChannelId, TestBitcoinCoreClient, TestDatabases, TestFeeEstimator}
+import org.scalatest.Assertions
+
+import java.net.InetAddress
+import java.util.UUID
+import java.util.concurrent.atomic.AtomicLong
+import scala.concurrent.duration.DurationInt
+import scala.util.{Random, Try}
+
+
+/**
+ * A minimal node setup, with real actors.
+ *
+ * Only the bitcoin watcher and wallet are mocked.
+ */
+case class MinimalNodeFixture private(nodeParams: NodeParams,
+                                      system: ActorSystem,
+                                      register: ActorRef,
+                                      router: ActorRef,
+                                      relayer: ActorRef,
+                                      switchboard: ActorRef,
+                                      paymentInitiator: ActorRef,
+                                      paymentHandler: ActorRef,
+                                      watcher: TestProbe,
+                                      wallet: DummyOnChainWallet)
+
+object MinimalNodeFixture extends Assertions {
+
+  def nodeParamsFor(alias: String, seed: ByteVector32): NodeParams = {
+    NodeParams.makeNodeParams(
+      config = ConfigFactory.load().getConfig("eclair"),
+      instanceId = UUID.randomUUID(),
+      nodeKeyManager = new LocalNodeKeyManager(seed, Block.RegtestGenesisBlock.hash),
+      channelKeyManager = new LocalChannelKeyManager(seed, Block.RegtestGenesisBlock.hash),
+      torAddress_opt = None,
+      database = TestDatabases.inMemoryDb(),
+      blockHeight = new AtomicLong(400_000),
+      feeEstimator = new TestFeeEstimator
+    )
+      .modify(_.alias).setTo(alias)
+      .modify(_.chainHash).setTo(Block.RegtestGenesisBlock.hash)
+      .modify(_.routerConf.routerBroadcastInterval).setTo(1 second)
+      .modify(_.peerConnectionConf.maxRebroadcastDelay).setTo(1 second)
```

Comment:
This indenting is a bit weird...

Reply:
Did differently in 3d42e3af8, but now lines are not aligned 🙈 .